src/org/traccar/protocol/CarcellProtocolDecoder.java

```diff
@@ -0,0 +1,162 @@
+            .number("(dd)(dd).(dddd),")          // latitude
+            .expression("([EW])")
+            .number("(ddd)(dd).(dddd),")         // longitude
+            .number("(d+),")                     // speed
+            .number("(d+),")                     // course
+            .number("([-+]ddd)([-+]ddd)([-+]ddd),")       // x,y,z
+            .number("(d+),")                     // battery
+            .number("(d+),")                     // csq
+            .number("(d),")                      // jamming
+            .number("(d+),")                     // hdop
+            .expression("([CG])")                // clock type
+            .number("(dd)(dd)(dd),")             // date
+            .number("(dd)(dd)(dd),")             // time
+            .number("(d),")                      // block
+            .number("(d),")                      // ignition
+            .number("(d),")                      // cloned
+            .expression("([AF])")                // panic
+            .number("(d),")                      // painel
+            .number("(d+),")                     // battery voltage
+            .any()                               // full format
+            .compile();
+    
+    private static final Pattern PATTERN_CEL = new PatternBuilder()
+            .expression("([$%])")                // memory flag
+            .number("(d+),")                     // imei
+            .text("CEL,")
+            .expression("([NS])")
+            .number("(d+.d+),")                  // latitude
+            .expression("([EW])")
+            .number("(d+.d+),")                  // longitude
+            .number("(d+),")                     // speed
+            .number("(d+),")                     // course
+            .number("([-+]ddd)([-+]ddd)([-+]ddd),")       // x,y,z
+            .number("(d+),")                     // battery
+            .number("(d+),")                     // csq
+            .number("(d),")                      // jamming
+            .number("(d+),")                     // hdop
+            .expression("([CG])")                // clock type
+            .number("(dd)(dd)(dd),")             // date
+            .number("(dd)(dd)(dd),")             // time
+            .number("(d),")                      // block
+            .number("(d),")                      // ignition
+            .number("(d),")                      // cloned
+            .expression("([AF])")                // panic
+            .number("(d),")                      // painel
+            .number("(d+),")                     // battery voltage
+            .any()                               // full format
+            .compile();
+
+    @Override
+    protected Object decode(
+            Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        Pattern pattern = null;
+        String sentence = (String) msg;
+        
+        if ( sentence.indexOf("CEL,") < 0 ) {
```

Comment:
Please follow same code style as everywhere else. There should be no spaces after and before parentheses.